src/lib/dlt_user.c

```diff
@@ -1308,6 +1306,15 @@ DltReturnValue dlt_register_app(const char *apid, const char *description)
 
 #ifdef DLT_TRACE_LOAD_CTRL_ENABLE
     strncpy(trace_load_settings[0].apid, dlt_user.appID, DLT_ID_SIZE);
```

Comment:
```suggestion
    pthread_rwlock_wrlock(&trace_load_rw_lock);
    strncpy(trace_load_settings[0].apid, dlt_user.appID, DLT_ID_SIZE);
    pthread_rwlock_unlock(&trace_load_rw_lock);
```

Writing settings needs protection to prevent seg faults